plugins/scaffolder-backend/src/service/router.test.ts

```diff
@@ -120,11 +137,26 @@ describe('createRouter', () => {
       logger: getVoidLogger(),
       config: new ConfigReader({}),
       database: createDatabase(),
-      catalogClient: createCatalogClient(template),
+      catalogClient,
       reader: mockUrlReader,
       taskBroker,
     });
     app = express().use(router);
+
+    jest
+      .spyOn(catalogClient, 'getEntityByRef')
```

Comment:
the catalog client above already has a jest.fn here, no need to spy - but it's because of typescript isn't it? so it gets awkward to cast it to a mocked version of a client just to get access to getEntityByRef.mockImplementation

well, no biggie, just thought i'd mention